### PR TITLE
Update Landing Hero

### PR DESCRIPTION
**Pull request by rogermparent**
https://user-images.githubusercontent.com/9111807/211942023-6aabcc76-e4ed-4e1b-808c-6f8b33be4eb0.mp4

This PR changes the hero for dvc.org into one similar to mlem.ai, with three selectable terminal animations powered by typed.js.

I opted to write a new typed.js component here so we could get another iteration on it, but I did learn some lessons from mlem.ai's copy!

This new hero also has a feature that mlem.ai's doesn't: autoplay! Until the user has clicked one of the hero items, the next item will automatically be played two seconds after the current one finishes. Once the user has clicked an item, the auto-play behavior is disabled until next page visit.

### CODE REVIEW

File: src/components/Home/LandingHero/index.tsx [hunk 1]
```diff
@@ -1,106 +1,175 @@
...
+──────────────────────────────────────────────────────────────────────────
+`
+  }
+].map(item => {
+  const { terminal } = item
+  return {
+    ...item,
+    terminal: terminal.split('\n').map(wrapOutputLine).join('\n')
+  }
+})
```
Comment (rogermparent) [inline src/components/Home/LandingHero/index.tsx:57]: This preprocessing should be moved into build time

**Comment by dberenbaum** [comment]
Thanks @rogermparent! The component works well so far for me.

Some comments:
* Most importantly: we are missing the subheading under "(Not Just) Data Version Control" (it should say "Git-based data science. Apply version control to machine learning development, make your repo the backbone of your project, and instill best practices across your team.")
* Can we add some color to "(Not Just) Data Version Control" like in mlem.ai or otherwise make it pop more?
* For the code component, can we add window controls and syntax highlighting like in mlem.ai?
* Is it possible to slow down the output of `dvc exp run`? The idea is to show that each stage is checked one at a time. Sorry, I know this might be beyond what we can do and something better for an asciinema video.

**Comment by rogermparent** [comment]
Thanks for the feedback!

>     * Most importantly: we are missing the subheading under "(Not Just) Data Version Control" (it should say "Git-based data science. Apply version control to machine learning development, make your repo the backbone of your project, and instill best practices across your team.")

My bad, I was so focused on the slides that I forgot about the subheading!

>     * Can we add some color to "(Not Just) Data Version Control" like in mlem.ai or otherwise make it pop more?

Sure! Any ideas? I suppose I can look for other gradients around dvc.org.

>     * For the code component, can we add window controls and syntax highlighting like in mlem.ai?

Yep! The window controls would just be a simple background change, syntax highlighting may be a little more rough but if mlem.ai does it we can do it here as well.

>     * Is it possible to slow down the output of `dvc exp run`? The idea is to show that each stage is checked one at a time. Sorry, I know this might be beyond what we can do and something better for an asciinema video.

Yup! We can add arbitrary pauses in the terminal output at any point.

**Comment by rogermparent** [comment]
Alright! Just pushed up fixes for this round of feedback:

> Most importantly: we are missing the subheading under "(Not Just) Data Version Control" (it should say "Git-based data science. Apply version control to machine learning development, make your repo the backbone of your project, and instill best practices across your team.")
  
Added this!

> Can we add some color to "(Not Just) Data Version Control" like in mlem.ai or otherwise make it pop more?
  
Made this heading our brand purple color, just starting based off the miro

> For the code component, can we add window controls and syntax highlighting like in mlem.ai?

I added our `dvc` highlighter, but it's meant for dark background terminals like our docs so I switched the whole terminal to our docs code block colors.

image

> Is it possible to slow down the output of dvc exp run? The idea is to show that each stage is checked one at a time. Sorry, I know this might be beyond what we can do and something better for an asciinema video.

I refined the ability to add custom per-line pauses at the end of content, and added a 500ms pause before each of those lines.

**Comment by jendefig** [comment]
Can "(Not Just) Data Version Control" be slightly bigger and bolder?

**Comment by jendefig** [comment]
It looks great @rogermparent 

@dberenbaum @jorgeorpinel Given the most recent feedback we've received from Francesco Calcavecchia and in regards to banking customers (don't even remember where I read this at the moment), I think we missed the importance of claiming "open-source" in the subtitle.  We may need to work that in.

"Open-source, Git-based data science. Apply version control to machine learning development, make your repo the backbone of your project, and instill best practices across your team."

I like this, but I like the impact of the short sentence without it. 

"Git-based data science.  DVC is an open-source tool to help you apply version control to machine learning development, make your repo the backbone of your project, and instill best practices across your team."

Too long?

**Comment by rogermparent** [comment]
Personally, I really like "Open-source, Git-based data science." I think those two are of equally great importance to what sets us apart from current and other possible competitors.

**Comment by rogermparent** [comment]
> Can "(Not Just) Data Version Control" be slightly bigger and bolder?

Done! I made the title semibold (one step up in boldness) and scale with screen width, the max size being one step up from the current one.

**Comment by jorgeorpinel** [comment]
I like how it sounds and it's not too long (in a sentence, not as a subtitle). BUT while DVC is open source, users data science/ ML projects don't have to be, so it may be misleading? I guess that would apply to "Git-based" too but in that case it's clear that it refers to the workflow, I think. WDYT @dberenbaum

**Comment by jorgeorpinel** [comment]
Maybe DS -> ML workflows

**Comment by dberenbaum** [comment]
Looks great @rogermparent! I'm good with "Open-source, Git-based data science." @jorgeorpinel I think people will understand what we mean even if it could technically mean something else.

One more ask: I know we are trying to squeeze a lot (too much?) content in here. Could we look for ways to limit the space it takes, like reducing the margin above "(Not Just) Data Version Control"? On my laptop, the hero now expands beyond one screen, and on mobile I see a lot of empty space at the top.

**Comment by rogermparent** [comment]
> Could we look for ways to limit the space it takes, like reducing the margin above "(Not Just) Data Version Control"? On my laptop, the hero now expands beyond one screen, and on mobile I see a lot of empty space at the top.

Great point! I added a lot of extra padding to match the old hero, but the new content does turn the whole thing very tall. I scrunched it down a bit to make the whole thing fit in one screen height without being too cramped IMO:

1024x768
image

1280x800
image

1680x1050 (my native)
image

**Comment by dberenbaum** [comment]
@rogermparent Looks good!

I noticed you have some green highlighting of the `dvc exp show` command which I don't see in the deployed version:

Screenshot 2023-01-12 at 2 05 56 PM

The table itself has some formatting (bold text and colors) that would be great to include if possible:

Screenshot 2023-01-12 at 2 10 22 PM

Also, can we include this updated table instead since we have redone the experiment names to make them more readable? Here's the text output to make it easier:

```
 ───────────────────────────────────────────────────────────────────────────
  Experiment                 roc_auc.train   roc_auc.test   train.min_split
 ───────────────────────────────────────────────────────────────────────────
  workspace                        0.96464        0.93989   0.04
  bigrams-experiment               0.98667        0.94602   0.01
  ├── 0417ff0 [fuggy-zack]         0.98667        0.94602   0.01
  ├── cb5c648 [fishy-mene]         0.97652        0.94309   0.02
  ├── c0dca21 [atrip-teds]         0.96957         0.9407   0.03
  ├── 3d2c9c7 [brisk-aged]         0.96464        0.93989   0.04
  └── 6e910c4 [chirk-main]         0.96193        0.93979   0.05
 ───────────────────────────────────────────────────────────────────────────
```

**Comment by rogermparent** [comment]
> I noticed you have some green highlighting of the `dvc exp show` command which I don't see in the deployed version:

Cache issue, fixed now!

> The table itself has some formatting (bold text and colors) that would be great to include if possible:
> 
> Screenshot 2023-01-12 at 2 10 22 PM

This'll be a little more difficult, but it can be put in. It will require more manual control, though, as the current code highlighting is applied by our actual code highlighter, and that actual code highlighter's performance on DVC tables is a far cry from the terminal, notably not including any bold

image

> Also, can we include this updated table instead since we have redone the experiment names to make them more readable? Here's the text output to make it easier:

Copied it in!

**Review by dberenbaum — APPROVED** [review]
Great work @rogermparent!

**Comment by dberenbaum** [comment]
> I'll see if I can do a more manual solution because solving this with the prism highlighter is a task unto itself.

Thanks! I'm approving, but if we can get any table formatting included, all the better!

**Comment by rogermparent** [comment]
Added highlighting!

image

Manual formatting isn't very compatible with Prism, but changing Prism to only apply to the lines with commands fits perfectly fine with our intended purpose here!

Main downside is that the source in json is a bit rough to work with directly since newlines aren't supported in JSON.

File: src/components/Home/LandingHero/Typed/index.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  typedOptions,
+  className
+}: {
+  typedOptions: TypedOptions
+  className?: string
+}) => {
+  const el = useRef<HTMLPreElement | null>(null)
+  const typed = useRef<Typed | null>(null)
+
+  console.log(typedOptions?.strings)
```
Comment (yathomasi) [inline src/components/Home/LandingHero/Typed/index.tsx:17]:
```suggestion
```

**Review by yathomasi — APPROVED** [review]
Great work @rogermparent 🎉 . Everything looks good and already should be enough for 1st iteration.

There are a few things we might want to improve on next iteration:
1. There is a slight shift on the selected tab (must be due to the border) and also we might want to add a shadow for users to know it's a clickable button. We already have them working on mlem.ai. We can use them.
2. Just an idea on top of my head - We could have the data in markdown format
	- Upsides
		- Reusing existing code(maintainable)
		- we can have the data in an easy format (easy to read and maintainable) (eg: [dvctable](https://github.com/iterative/dvc.org/blob/1a94a86cadca02f50c3dc549459020e32a82f876/content/docs/use-cases/experiment-tracking.md?plain=1#L28-L41)
		- also, it will be parsed with a gatsby remark with all the related plugins applied.
		- Also, have these on the theme itself so that they can be reused on the multiple projects
	- Possible issue
		- Using it inside the type.js component
		- huge time effort

**Comment by rogermparent** [comment]
> There are a few things we might want to improve on next iteration:
> 
>   1. There is a slight shift on the selected tab (must be due to the border) and also we might want to add a shadow for users to know it's a clickable button. We already have them working on mlem.ai. We can use them.
> 
>   2. Just an idea on top of my head - We could have the data in markdown format

I had considered that while making this, it would actually be pretty easy using the same techniques as this PR- we just manually use the `File` node from Gatsby instead of `gatsby-transformer-json` so we could drop in any markdown parser pretty seamlessly

>  * Reusing existing code(maintainable)

It'll probably need new code, but not too much.

>  * we can have the data in an easy format (easy to read and maintainable) (eg: [dvctable](https://github.com/iterative/dvc.org/blob/1a94a86cadca02f50c3dc549459020e32a82f876/content/docs/use-cases/experiment-tracking.md?plain=1#L28-L41)

>  * also, it will be parsed with a gatsby remark with all the related plugins applied.
>  * Also, have these on the theme itself so that they can be reused on the multiple projects

>  Possible issue: * Using it inside the type.js component

Typed is surprisingly easy to put anything into, the real problem is Prism which automatically escapes all  when used normally, and I can't find any ability to prevent that behavior. Now that I type that out and think about it, maybe putting Prism earlier in the pipeline would do the trick?

>  * huge time effort

There are some quick fixes that could give big benefits. Swapping JSON for YAML or MD in the simplest form would be pretty quick and those formats would support newlines and be less horrible to work with than JSON currently is.

**Comment by jendefig** [comment]
Heads up! 
The auto-playing feature isn't working.  This is how the screen is staying right now. 
image

**Pull request merged**